distributed/tests/test_s3.py

```diff
@@ -0,0 +1,91 @@
+from contextlib import contextmanager
+
+import boto3
+from dask.imperative import Value
+from distributed import Executor
+from distributed.executor import _wait
+from distributed.s3 import read_contents, get_list_of_summary_objects, read_content_from_keys
+from distributed.utils import get_ip
+from distributed.utils_test import gen_cluster
+
+ip = get_ip()
+test_bucket_name = 'distributed-test'
+
+@contextmanager
+def make_s3(bucket_name):
+    s3 = boto3.resource('s3')
+    bucket = s3.create_bucket(Bucket=bucket_name)
+
+    try:
+        yield bucket
+    finally:
+        for key in bucket.objects.all():
+            key.delete()
+        bucket.delete()
+
+
+def test_get_list_of_summary_objects():
+    with make_s3(test_bucket_name) as bucket:
+        data = b'a' * int(10)
+        fn_1 = 'tmp/test/file1'
+        fn_2 = 'tmp/test/file2'
+
+        bucket.put_object(Key=fn_1, Body=data)
+        bucket.put_object(Key=fn_2, Body=data)
+
+        L = get_list_of_summary_objects(test_bucket_name, prefix='tmp')
+
+        assert len(L) == 2
+        assert map(lambda o: o.key, L) == ['tmp/test/file1', 'tmp/test/file2']
```

Comment:
hmm, come to think of it, i also need to add delimiter here as well. prefix + delimiter should cover most of the hierarchical keys use cases.  More tests coming up.